sshslot.py

```diff
@@ -54,6 +54,24 @@ def get_slots(self):
     def get_name(self):
         return self.host
 
+class SlotProcess:
+    def __init__(self, log):
+        self.p = None
+        self.can_kill = threading.Event()
```

Comment:
I'm a bit concerned we could leak kill threads this way - this event might never happen if something leading up to the final shell exec fails. Simple fix would be a timeout, more complex would be to set this event in the failure places.